src/db2ia/dbstmt.cc

```diff
@@ -2609,16 +2609,16 @@ int DbStmt::bindParams(Napi::Env env, Napi::Array *params, std::string &error)
         size_t str_length = string.length();
         const char *cString = string.c_str();
         param[i].valueType = SQL_C_CHAR;
-
-        if (param[i].io == SQL_PARAM_INPUT || param[i].io == SQL_PARAM_INPUT_OUTPUT)
-        {
-          param[i].buf = strdup(cString);
-          param[i].ind = str_length;
-        }
-        else if (param[i].io == SQL_PARAM_OUTPUT)
-        {
-          param[i].buf = (char *)calloc(param[i].paramSize + 1, sizeof(char));
-          param[i].ind = param[i].paramSize;
+        // CLI does not honor the buffer size parameter or the output size in the indicator
+        // Make the buffer size is at least the size of parameter or the size of the string
+        param[i].buf = (char *)calloc(std::max(static_cast<size_t>(param[i].paramSize) + 1, str_length), sizeof(char));

```

Comment:
str_length will be without the null terminator too
```suggestion
        param[i].buf = (char *)calloc(std::max(static_cast<size_t>(param[i].paramSize), str_length) + 1, sizeof(char));

```

Reply:
Updated